mirrord-layer/src/go_hooks.rs

```diff
@@ -1,468 +1,659 @@
+        "mov    rcx, rax",
+        "mov    rax, -0x1",
+        "mov    rbx, 0x0",
+        "xorps  xmm15, xmm15",
+        "mov    r14, QWORD PTR FS:[0xfffffff8]",
+        "ret",
+        "2:",
+        // RAX already contains return value
+        "mov    rbx, 0x0",
+        "mov    rcx, 0x0",
+        "xorps  xmm15, xmm15",
+        "mov    r14, QWORD PTR FS:[0xfffffff8]",
+        "ret",
+        options(noreturn)
+    )
+}
+
+/// Hooks for when hooking a pre go 1.19 binary
+fn pre_go1_19(interceptor: &mut Interceptor, binary: &str) {
+    hook_symbol!(
+        interceptor,
+        "syscall.RawSyscall.abi0",
+        go_rawsyscall_detour,
+        binary
+    );
+    hook_symbol!(
+        interceptor,
+        "syscall.Syscall6.abi0",
+        go_syscall6_detour,
+        binary
+    );
+    hook_symbol!(
+        interceptor,
+        "syscall.Syscall.abi0",
+        go_syscall_detour,
+        binary
+    );
+}
 
-    /// Note: We only hook "RawSyscall", "Syscall6", and "Syscall" because for our usecase,
-    /// when testing with "Gin", only these symbols were used to make syscalls.
-    /// Refer:
-    ///   - File zsyscall_linux_amd64.go generated using mksyscall.pl.
-    ///   - https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/syscall/syscall_unix.go
-    pub(crate) fn enable_socket_hooks(interceptor: &mut Interceptor, binary: &str) {
-        hook_symbol!(
-            interceptor,
-            "syscall.RawSyscall.abi0",
-            go_rawsyscall_detour,
-            binary
-        );
-        hook_symbol!(
-            interceptor,
-            "syscall.Syscall6.abi0",
-            go_syscall6_detour,
-            binary
-        );
-        hook_symbol!(
-            interceptor,
-            "syscall.Syscall.abi0",
-            go_syscall_detour,
-            binary
-        );
+/// Hooks for when hooking a post go 1.19 binary
+fn post_go1_19(interceptor: &mut Interceptor, binary: &str) {
+    hook_symbol!(
+        interceptor,
+        "runtime/internal/syscall.Syscall6",
+        go_syscall_new_detour,
+        binary
+    );
+}
+
+/// Note: We only hook "RawSyscall", "Syscall6", and "Syscall" because for our usecase,
+/// when testing with "Gin", only these symbols were used to make syscalls.
+/// Refer:
+///   - File zsyscall_linux_amd64.go generated using mksyscall.pl.
+///   - https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/syscall/syscall_unix.go
+pub(crate) fn enable_socket_hooks(interceptor: &mut Interceptor, binary: &str) {
+    if let Some(version_symbol) =
+        frida_gum::Module::find_symbol_by_name(binary, "runtime.buildVersion.str")
+    {
+        // Version str is `go1.xx` - take only last 4 characters.
+        let version = unsafe {
+            std::str::from_utf8_unchecked(std::slice::from_raw_parts(
+                version_symbol.0.add(2) as *const u8,
+                4,
+            ))
+        };
+        let version_parsed: f32 = version.parse().unwrap();
```

Comment:
this might crash on non numbered versions of go, example is with Experiment flags enabled

Reply:
mm okay sounds like an edge-enough case to not support atm